File: HW/HW3/Main.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.print("Enter an array of integers separated by a space: ");
        String str = in.nextLine();

        String[] strNums = str.split(" ");
        int[] nums = new int[strNums.length];
        for(int i = 0; i < strNums.length; i++) {
            nums[i] = Integer.parseInt(strNums[i]);
        }

        for(int i = 1; i < nums.length; i++) {
            int item = nums[i];
            int j = i - 1;
            while(j >= 0 && nums[j] > item) {
                nums[j+1] = nums[j];
                nums[j] = item;
                j -= 1;
            }
        }
        for (int num : nums) {
            System.out.print(num + " ");
        }
    }
}

File: HW/HW3/main.py


def insSort(array):
    integers = True
    try:
        float(array[0])
    except ValueError:
        integers = False

    if integers:
        for i in range(0, len(array)):
            array[i] = float(array[i])

    for i in range(1, len(array)):
        item = array[i]
        j = i - 1
        while j >= 0 and array[j] > item:
            array[j+1] = array[j]
            array[j] = item
            j -= 1
    return array


Input = input("Enter array of words or numbers separated by a space: ")
array = Input.split(" ")
array = insSort(array)
print(array)
